docs/testing.rst

```diff
@@ -81,3 +81,76 @@ When the reason for the breackage is identified there are two options to go forw
     * If the failure comes from an intended change in behaviour adjust Anaconda or the tests
     * If the failure uncovers an actual regression, file bugs for the given Fedora components. If it does not look like the issue will be
       fixed quickly, work around in Anaconda or add a `naughty override file <https://github.com/cockpit-project/bots/tree/main/naughty/>`_, thus marking the expected failure pattern.
+
+Fix Web UI pixel tests
+----------------------
+
+So pixel tests fail on your PR - now what ?
+
+There are essentially two options:
+
+    * the test uncovered a in your PR bug that needs to be fixed in the code
+    * your PR changes the visual behavior of the Web UI in a valid manner and the pixel test needs to be fixed
+
+We will cover the second option.
+
+First make sure your PR is rebased on latest revision of the target branch. This is important as the issue you are seeing might have already been fixed by another PR that has since been merged.
+
+If rebasing your PR did not fix the issue, and pixel tests still fail in CI on your PR, you will need to update one or more of the reference images.
+
+::
+
+    $ cd ui/webui
+    $ GITHUB_BASE=rhinstaller/anaconda ./test/common/pixel-tests pull
+
+Next find the failing test in the PR CI test results and find the individual test that is failing. The results page shows an image comparison tool and in its header links to the new screenshot that no longer matches the expected picture:
+
+
+::
+
+    New <something>-fail-pixels.png on the left, reference on the right.
+
+Download the screenshot into this screenshot into the `ui/webui/test/reference` folder and check it replaced an existing file, as the new screenshot by definition will be named the same as one of the existing pictures that no longer match. For example:
```

Comment:
"Download the screenshot into this screenshot" - Should this say "download the screenshot from the page and save with the right name into the ... directory" ?

Reply:
Indeed, this was a typo - should be fixed now & I've also changed a bit the structure of the sentence.